src/main/java/com/minecolonies/entity/ai/citizen/guard/EntityAIMeleeGuard.java

```diff
@@ -0,0 +1,227 @@
+package com.minecolonies.entity.ai.citizen.guard;
+
+import com.minecolonies.colony.jobs.JobGuard;
+import com.minecolonies.entity.ai.util.AIState;
+import com.minecolonies.entity.ai.util.AITarget;
+import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.init.SoundEvents;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.ItemSword;
+import net.minecraft.item.ItemTool;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.EnumHand;
+import org.jetbrains.annotations.NotNull;
+
+import static com.minecolonies.entity.ai.util.AIState.*;
+
+/**
+ * Handles the AI of the guard entities.
+ */
+public class EntityAIMeleeGuard extends AbstractEntityAIGuard
+{
+    /**
+     * Basic delay for the next shot.
+     */
+    private static final int BASE_RELOAD_TIME = 30;
+
+    /**
+     * The pitch will be divided by this to calculate it for the arrow sound.
+     */
+    private static final double PITCH_DIVIDER = 1.0D;
+
+    /**
+     * The base pitch, add more to this to change the sound.
+     */
+    private static final double BASE_PITCH = 0.8D;
+
+    /**
+     * Random is multiplied by this to get a random arrow sound.
+     */
+    private static final double PITCH_MULTIPLIER = 0.4D;
+
+    /**
+     * Quantity to be moved to rotate the entity without actually moving.
+     */
+    private static final double MOVE_MINIMAL = 0.01D;
+
+    /**
+     * Quantity the worker should turn around all at once.
+     */
+    private static final double TURN_AROUND = 180D;
+
+    /**
+     * Normal volume at which sounds are played at.
+     */
+    private static final double BASIC_VOLUME = 1.0D;
+
+    /**
+     * Experience the guard receives each shot arrow.
+     */
+    private static final double XP_EACH_HIT = 0.2;
+
+    /**
+     * Base speed of the guard he follows his target.
+     */
+    private static final int BASE_FOLLOW_SPEED = 1;
+
+    /**
+     * Base multiplier increasing the attack speed each level.
+     */
+    private static final double BASE_FOLLOW_SPEED_MULTIPLIER = 0.25D;
+
+    /**
+     * The Min distance of the guard to attack entities.
+     */
+    private static final double MIN_ATTACK_DISTANCE = 2.0D;
+
+    /**
+     * Damage per range attack.
+     */
+    private static final double DAMAGE_PER_ATTACK = 0.5;
+
+    /**
+     * Chance that a mob is lit on fire when a weapon has the fire aspect enchantment.
+     */
+    private static final int FIRE_CHANCE_MULTIPLIER = 4;
+
+    /**
+     * Sets up some important skeleton stuff for every ai.
+     *
+     * @param job the job class
+     */
+    public EntityAIMeleeGuard(@NotNull final JobGuard job)
+    {
+        super(job);
+        super.registerTargets(
+                new AITarget(IDLE, () -> START_WORKING),
```

Comment:
Stuff like IDLE->START_WORKING can be defined in the upper class. You can just go from the top and move in all methods you can move in.

If they later seem to differ, we can pull them back up